XRTK-Core/Packages/com.xrtk.core/Definitions/Devices/ControllerMappingLibrary.cs

```diff
@@ -219,93 +219,96 @@ public static class ControllerMappingLibrary
         /// </summary>
         /// <param name="mappingProfile"></param>
-        /// <param name="controllerType"></param>
-        /// <param name="handedness"></param>
-        /// <returns>The scaled texture for the controller type, if none found then a generic texture is returned.</returns>
-        public static Texture2D GetControllerTextureScaled(BaseMixedRealityControllerMappingProfile mappingProfile, SupportedControllerType controllerType, Handedness handedness)
+        /// <returns>The scaled texture for the controller profile, if none found then a generic texture is returned.</returns>
+        /// <remarks>
+        /// The file name should be formatted as:<para/>XRTK/StandardAssets/Textures/{ControllerName}_{handedness}_{theme}_{scaled}.png<para/>
+        /// </remarks>
+        public static Texture2D GetControllerTextureScaled(MixedRealityControllerMappingProfile mappingProfile)
         {
-            return GetControllerTextureCached(mappingProfile, controllerType, handedness, true);
+            return GetControllerTextureCached(mappingProfile, true);
         }
 
-        private static Texture2D GetControllerTextureCached(BaseMixedRealityControllerMappingProfile mappingProfile, SupportedControllerType controllerType, Handedness handedness, bool scaled = false)
+        private static Texture2D GetControllerTextureCached(MixedRealityControllerMappingProfile mappingProfile, bool scaled = false)
         {
-            var key = new Tuple<SupportedControllerType, Handedness, bool>(controllerType, handedness, scaled);
+            var key = new Tuple<Type, Handedness, bool>(mappingProfile.ControllerType.Type, mappingProfile.Handedness, scaled);
 
             if (CachedTextures.TryGetValue(key, out var texture))
             {
                 return texture;
             }
 
-            texture = GetControllerTextureInternal(mappingProfile, controllerType, handedness, scaled);
+            texture = GetControllerTextureInternal(mappingProfile, scaled);
             CachedTextures.Add(key, texture);
             return texture;
         }
 
-        private static Texture2D GetControllerTextureInternal(BaseMixedRealityControllerMappingProfile mappingProfile, SupportedControllerType controllerType, Handedness handedness, bool scaled)
+        private static Texture2D GetControllerTextureInternal(MixedRealityControllerMappingProfile mappingProfile, bool scaled)
         {
             if (mappingProfile != null &&
-                mappingProfile.ControllerType == controllerType)
+                mappingProfile.ControllerType.Type != null)
             {
-                var texture = GetControllerTextureInternal(mappingProfile.TexturePath, handedness, scaled);
+                var texture = GetControllerTextureInternal($"{PathFinderUtility.XRTK_Core_RelativeFolderPath}/StandardAssets/Textures/{mappingProfile.ControllerType.Type.Name}", mappingProfile.Handedness, scaled);
```

Comment:
This now only gets controller images by the name of the type. We should figure out a smart way to reference multiple types assigned to a single image.